### PR TITLE
Temporal Upscaling + some base TAA improvements

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
Gonna be a vaguely long PR description, so 

TL;DR: 

- Temporal Upscaling algorithm for dynamic resolution. 
- Take into account jittering vector when computing filter weights for TAA high quality.
- Reduce flickering by a bit (no need to remove the jitter on the reprojection coordinates) 
- Exposing a base blend factor parameter to more/less history (as advanced setting), this can help reduce ghosting in favour of more aliasing or viceversa. 
- Removed some ringing from sharpening filter.
- Optimized also DLSS path by sampling the original resolution input textures (depth and motion vectors) instead of having to upscale them (their upscale doesn't contribute quality, I checked with Kleber and it was for simplicity). 

--------------------- 

## Temporal Upscaling 

Added an option to perform temporal upscaling as Dynamic resolution upscaling filter. Similarly to DLSS it happens in place of TAA. 

Specifically this pass is a modification of the TAA shader, most notable differences is to map the incoming input into an output resolution, the resampling strategy is a very sharp gaussian filter (see For Honor XFest presentation) and a confidence filter at the end for accumulating the history only on pixels we are confident about. 
Generally there is a bit more of details, but the gist is the above. 

The variance clipping corners are also a bit contracted as acting on lower resolutions will lead to possibly more ghosting. 

### Visual results

Full resolution images are available upon requests.  

**The following 2 images are with a zoom into the image**
 I suggest opening in a new tab, also note that even though representative, slight quality improvements have been made since I took the images (so actual result is _slightly_ better) 

Comparison

Comparison2

Comparison including also DLSS with 50% scale (25% amount of pixels). DLSS quality is still better (I suspect mostly due to better sharpening), but the quality is in that ballpark according to first tests with QA.

https://user-images.githubusercontent.com/43168857/124601519-36a9e100-de68-11eb-813e-55c37cb7f2c1.mp4

**Native 4k** (sadly compressed)
SampleScene_20210702_1811422_Native

**Catmull-Rom 70% (50% amount of pixels)**
SampleScene_20210702_1811776_Dynamic Resolution - 70 - Catmull Rom

**DLSS 70% (50% amount of pixels)** 
SampleScene_20210702_1812413_Dynamic Resolution - 70 - DLSS

**TAAU 70% (50% amount of pixels)** 
SampleScene_20210702_1811930_Dynamic Resolution - 70 - TAA-U

Uncompressed images https://drive.google.com/drive/u/1/folders/130RVIBqFfpuwKyiWb6ipPK76203x9OO4

I have many more images if desired. 

Random silly one with very low scale to have a more dramatic comparison
image

### Performance impact

The cost of the pass is very similar to TAA high quality, the actual cost depends on how big the output resolution is. It is roughly similar (though a bit cheaper) than running said TAA high quality on the destination output. 

**PS4 Pro**  

VERY BASIC tests on template (I don't have a pro devkit, this was provided by Jans)

| **GPU COST** | **Native 4K**      | **TAAU 70% (~50% pixels)** | **TAAU 50% (~25% pixels)** |
|----------|----------------|------------------------|------------------------|
|          | ~60ms (15 FPS) | ~36ms (~28FPS)         | ~22ms (~46FPS)         |

**PC**

Of course PC numbers are what they are, but @remi-chapelain  helped getting some numbers. The PC in question is fairly beefy, so more numbers are needed to be collected. 

The template shot above

|             | Native 4K         | 70%                | 50%                |
|-------------|-------------------|--------------------|--------------------|
| Catmull-Rom | 39.2ms (~25.5FPS) | 15ms (~66.6 FPS)   | 8.75ms (~114FPS)   |
| TAA Upscale |     39.2ms (~25.5FPS)        | 15.3ms (~65.3 FPS) | 10.56ms (~94.7FPS) |
| DLSS        |      39.2ms (~25.5FPS)         | ~16.3 (~61.3 FPS)  | ~10.85ms (~92 FPS) |

Spaceship benchmark (1440p, 4K hopefully to come. Here differences are less dramatic as bottlenecks with lower start resolution shift). 

|             | Native 1440p         | 70%                | 50%                |
|-------------|-------------------|--------------------|--------------------|
| Catmull-Rom | 11.5ms (86.6FPS) | 7.98ms (125.3FPS) | 7.72ms (129.5FPS) |
| TAA Upscale |  11.5ms (86.6FPS)| 8.26 (121.1 FPS) | 7.6 (131.5FPS) |
| DLSS        |      11.5ms (86.6FPS)         |  8.74 (114.4 FPS)  | 7.61 (131.4 FPS) |

### Known issues / missing things

- Currently Exclude from TAA stencil bit don't work in this pass, even if excluded from TAA it will go through TAAU. 
- Missing expanded documentation and guidelines (e.g. speed rejection with low percentages kinda work bad)
- A bit more exploration on confidence factor can be done. 
- A test. 

Generally another iteration wouldn't necessarily hurt.  I'd like to get some feedback first, hence the (Experimental) 

## Normal TAA changes 

This PR also introduces a couple of TAA improvements. 

**Fix TAA High quality filtering**
Finally taking jittering into account, on screenshots here might be subtle, if interested I suggest opening in separate tabs and switch back and forth. 

**Before**
image
**After**
image

**Expose base blend factor**

To control how much history is taken each frame, the trade off is aliasing vs ghosting. 

(NOTE: Ghosting is exaggerated in gifs, this way just to show the parameter impact)

https://user-images.githubusercontent.com/43168857/124593363-d82c3500-de5e-11eb-9b9c-9c3f8b5a6106.mp4

Such parameter is only in advanced parameters since I don't expect non-power user need to touch this. 

**Reducing ringing in sharpening filter**

**Before**
OldSharpen

**After**
NewSharpen

____________________

Both for TAAU and normal TAA we can and should do an optimization pass (now that we can sample stencil in compute, we can move TAA to compute and leverage LDS) which should simplify a few things as well.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader [hunk 1]
```diff
@@ -141,45 +172,79 @@ Shader "Hidden/HDRP/TemporalAA"
...
-            DecodeMotionVector(LOAD_TEXTURE2D_X(_CameraMotionVectorsTexture, closest), motionVector);
+
+            DecodeMotionVector(SAMPLE_TEXTURE2D_X_LOD(_CameraMotionVectorsTexture, s_point_clamp_sampler, ClampAndScaleUVForPoint(uv + closestOffset * _InputSize.zw), 0), motionVector);
             // --------------------------------------------------------
 
             // --------------- Get resampled history ---------------
             float2 prevUV = input.texcoord - motionVector;
 
-            CTYPE history = GetFilteredHistory(_InputHistoryTexture, prevUV, _HistorySharpening, _TaaHistorySize);
+            CTYPE history = GetFilteredHistory(_InputHistoryTexture, prevUV, _HistorySharpening, _TaaHistorySize, _RTHandleScaleForTAAHistory);
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:220]: Maybe add a small comment explaining why we need a separate scale here.

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:225]: Added on top where the scales are defined

**Comment by kecho** [comment]
Hey I pulled your branch and its not quite working. Getting all sorts of edge artifacts on the HDRP template. All I did is enabeld DRS and then enabled TAA on the camera.
image

Another problem, if we disable TAA from the camera we also get just a partial part of the frame
image

**Review by kecho — CHANGES_REQUESTED** [review]
Pulled your branch and im seeing a bunch of artifacts. See screenshots in PR conversation.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs [hunk 2]
```diff
@@ -40,10 +44,40 @@ partial class Rendering
...
+
             static void Drawer_Rendering_AllowDynamicResolution(SerializedHDCamera p, Editor owner)
             {
                 CameraUI.Output.Drawer_Output_AllowDynamicResolution(p, owner);
 
+                var dynamicResSettings = HDRenderPipeline.currentAsset.currentPlatformRenderPipelineSettings.dynamicResolutionSettings;
+                s_IsRunningTAAU = p.allowDynamicResolution.boolValue && dynamicResSettings.upsampleFilter == UnityEngine.Rendering.DynamicResUpscaleFilter.TAAU && dynamicResSettings.enabled;
+
+                if (s_IsRunningTAAU)
+                    EditorGUILayout.HelpBox(Styles.taauInfoBox, MessageType.Info);
```
Comment (kecho) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:79]: This should probably get cleaned up:
Instead of failing when another TAA algorithm is chosen, turn off DRS / TAAU and instead in the warning say that dynamic resolution is disabled.
My point is that we shouldnt allow the pipeline to get into a corrupted config so easily, even if we add warnings.

**Review by kecho — COMMENTED** [review]
Figured out my first issue: its caused by having 2 views open, if I have the Scene view and gameview open at the same time, I see these artifacts, likely history always rejected. I recommend removing all global state (state inside the HDRenderingPipeline.PostProcess.cs to check last frame stuff) and instead 2 options:
* Add the state in the HDCamera
* Follow the same pattern with weak references I had on DLSSPass.cs

I like more the approach of DLSSPass.cs becuase we dont bloat the camera, its efficient and we can manage all cameras from there safely.

Its up to you of course

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader [hunk 3]
```diff
@@ -189,19 +259,22 @@ Shader "Hidden/HDRP/TemporalAA"
...
 
-#if ANTI_FLICKER_MV_DEPENDENT || VELOCITY_REJECTION
-            float motionVectorLength = length(motionVector);
-#else
             float motionVectorLength = 0.0f;
+            float motionVectorLenInPixels = 0.0f;
+
+#if ANTI_FLICKER_MV_DEPENDENT || VELOCITY_REJECTION
+            motionVectorLength = length(motionVector);
+            motionVectorLenInPixels = motionVectorLength * length(_InputSize.xy);
```
Comment (kecho) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:267]: good idea to precompute length(_InputSize.xy) and pass it as a constant

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader [hunk 4]
```diff
@@ -221,9 +294,12 @@ Shader "Hidden/HDRP/TemporalAA"
 
 #if VELOCITY_REJECTION
             float lengthMV = motionVectorLength * 10;
```
Comment (kecho) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:296]: Whats the meaning of that *10 for the motion vector length?

**Comment by kecho** [comment]
Review conclusion:
Pretty awesome work Francesco, once I got passed the view bugs got to play with it. Its mostly a struggle of blurry and stable vs jittery and sharp. Overall, an absolute must in terms of features to bridge our gap.
I recommend not even trying making this experimental. Work on the few (minor) issues highlighted here, do a test and just do another pass on docs after.

Cheers

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader [hunk 5]
```diff
@@ -141,45 +177,79 @@ Shader "Hidden/HDRP/TemporalAA"
...
 #if ORTHOGRAPHIC
-            float2 closest = input.positionCS.xy;
+            float2 closestOffset = 0;
 #else
-            float2 closest = GetClosestFragment(_DepthTexture, int2(input.positionCS.xy));
+            float2 closestOffset = GetClosestFragmentOffset(_DepthTexture, int2(input.positionCS.xy));
 #endif
-            DecodeMotionVector(LOAD_TEXTURE2D_X(_CameraMotionVectorsTexture, closest), motionVector);
+
+            DecodeMotionVector(SAMPLE_TEXTURE2D_X_LOD(_CameraMotionVectorsTexture, s_point_clamp_sampler, ClampAndScaleUVForPoint(uv + closestOffset * _InputSize.zw), 0), motionVector);
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:219]: Because we have UVs sampling it rather than specific locations and handling it with UV makes the scaling/clamping easier

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:79]: This is currently not failing though, it is just enforcing an option. It does not corrupt it just forces as visible the UX that is matching whatever the TAAU config requires. 

Unless I misunderstood? 

One issue that comes to mind though is if someone set say SMAA before and then enables TAAU I need to check what happens (i.e. if SMAA runs or not). I think in that case we should just skip other AA passes in the post process logic.

[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:79]: I see how actually I should do the enforcing in a different way, changes incoming

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs [hunk 6]
```diff
@@ -132,7 +166,7 @@ static void Drawer_Rendering_Antialiasing(SerializedHDCamera p, Editor owner)
             static CED.IDrawer AntialiasingModeDrawer(HDAdditionalCameraData.AntialiasingMode antialiasingMode, CED.ActionDrawer antialiasingDrawer)
             {
                 return CED.Conditional(
-                    (serialized, owner) => serialized.antialiasing.intValue == (int)antialiasingMode,
+                    (serialized, owner) => (serialized.antialiasing.intValue == (int)antialiasingMode) && (s_IsRunningTAAU ? serialized.antialiasing.intValue == (int)HDAdditionalCameraData.AntialiasingMode.TemporalAntialiasing : true),
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:169]: This was an attempt of doing an enforcing of an option, but I realise now there is a better way, gonna do that now.

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader [hunk 7]
```diff
@@ -141,45 +177,79 @@ Shader "Hidden/HDRP/TemporalAA"
...
 
             // --------------- Get closest motion vector ---------------
             float2 motionVector;
 
 #if ORTHOGRAPHIC
-            float2 closest = input.positionCS.xy;
+            float2 closestOffset = 0;
 #else
-            float2 closest = GetClosestFragment(_DepthTexture, int2(input.positionCS.xy));
+            float2 closestOffset = GetClosestFragmentOffset(_DepthTexture, int2(input.positionCS.xy));
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:216]: Indeed it should be using outputPixInInput, I did the move to offset for this reason and then forgot to update the right posSS D:

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader [hunk 8]
```diff
@@ -141,45 +177,79 @@ Shader "Hidden/HDRP/TemporalAA"
...
 #endif
-            DecodeMotionVector(LOAD_TEXTURE2D_X(_CameraMotionVectorsTexture, closest), motionVector);
+
+            DecodeMotionVector(SAMPLE_TEXTURE2D_X_LOD(_CameraMotionVectorsTexture, s_point_clamp_sampler, ClampAndScaleUVForPoint(uv + closestOffset * _InputSize.zw), 0), motionVector);
             // --------------------------------------------------------
 
             // --------------- Get resampled history ---------------
             float2 prevUV = input.texcoord - motionVector;
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:223]: Motion vector is indeed a different resolution, but it should be sampled using UV snapped at the center of the corresponding input position (see https://github.com/Unity-Technologies/Graphics/pull/5066/files/a29eb8eb958c92a952d19c8a4ce2b32038ab98a7#diff-8889d598aaabebeb6c5d4ced55e9f5cd8b435b869329725b7794fad915bb6460R207 ), so it's the closest we can get to the proper value. 

The prevUV in itself acts  on same resolution of the output (history has the same size), but the motion vectors unfortunately are not full res so the snapping is the best we can do.

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader (hunk 4 above)
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:296]: A mix of empirically set value and better space to store it in the limited bit range of the output history. 

I shall probably add a comment

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.PostProcess.cs [hunk 9]
```diff
@@ -66,10 +66,13 @@ private enum SMAAStage
         HableCurve m_HableCurve;
 
         //Viewport information
-
+        Rect m_PrevFinalViewport = new Rect(Vector2.zero, Vector2.one * -1);
         Vector2Int m_AfterDynamicResUpscaleRes = new Vector2Int(1, 1);
         Vector2Int m_BeforeDynamicResUpscaleRes = new Vector2Int(1, 1);
 
+        Dictionary<HDCamera, bool> m_WasRunningTAAU = new Dictionary<HDCamera, bool>();
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.PostProcess.cs:73]: They absolutely are :D

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.PostProcess.cs [hunk 10]
```diff
@@ -1428,21 +1447,46 @@ class TemporalAntiAliasingData
...
-            float crossWeights = Mathf.Exp(-2.29f * 2);
-            float plusWeights = Mathf.Exp(-2.29f);
-            float centerWeight = 1;
+            passData.taaParameters = new Vector4(TAAU && postDoF ? 0.25f : camera.taaHistorySharpening, postDoF ? maxAntiflicker : Mathf.Lerp(minAntiflicker, maxAntiflicker, camera.taaAntiFlicker), motionRejectionMultiplier, temporalContrastForMaxAntiFlicker);
 
-            float totalWeight = centerWeight + (4 * plusWeights);
-            if (camera.TAAQuality == HDAdditionalCameraData.TAAQualityLevel.High)
+            // Precompute weights used for the Blackman-Harris filter.
+
+            Vector2Int[] offsets = new Vector2Int[]
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.PostProcess.cs:1456]: I am not sure about this. The order of the offsets are at the moment really important; they need to match what the shader specifically expects due to the swizzling required caused by the quad communication. 

I struggle to see the usage  of this specific thing outside of this pass and it'd be called only once.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:79]: image

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:169]: https://github.com/Unity-Technologies/Graphics/pull/5066#discussion_r665132217

**Comment by FrancescoC-unity** [comment]
> Hey I pulled your branch and its not quite working. Getting all sorts of edge artifacts on the HDRP template. All I did is enabeld DRS and then enabled TAA on the camera.
> image
> 
> Another problem, if we disable TAA from the camera we also get just a partial part of the frame
> image

Argh, I should use the two camera layout  more often. Will fix that up

Re the second issue, the issue is that other AA are selectable, forced TAA now.

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader (hunk 3 above)
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader:273]: I  think at the time (I just pulled this out of a function here) I measured that the cost of fetching and waiting for the cbuffer was higher than the extra ALU work here. 

I need to do an optimization pass later anyway

**Comment by FrancescoC-unity** [comment]
Note; I found an issue with PB DoF, looking at it now.

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/DoFMinMaxDilate.compute [hunk 11]
```diff
@@ -34,7 +34,7 @@ void KMain(uint3 dispatchThreadId : SV_DispatchThreadID)
     float4 tileInfo = _InputTexture[COORD_TEXTURE2D_X(dispatchThreadId.xy)];
 
     int2 startIndex = max((int2)dispatchThreadId.xy - int2(1, 1), int2(0, 0));
-    int2 endIndex =  min((int2)dispatchThreadId.xy + int2(1, 1), _ScreenSize.xy / TILE_RES - int2(1, 1));
+    int2 endIndex =  min((int2)dispatchThreadId.xy + int2(1, 1), _PostProcessScreenSize.xy / TILE_RES - int2(1, 1));
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/DoFMinMaxDilate.compute:37]: This is the fix I mentioned to both of you in slack @pmavridis  (cc: @kecho as it is going to fix some issues with DLSS  too) 

(A similar change also in https://github.com/Unity-Technologies/Graphics/pull/5066/commits/c3fc9ff70fc6e887f022869811ffd35b241aae74#diff-615e5f83c56e2e6631a5ef1db8170921d0702eed82a4f08cec8c97f22456f8c5R33 )

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.PostProcess.cs [hunk 12]
```diff
@@ -2879,10 +2879,10 @@ TextureHandle DepthOfFieldPass(RenderGraph renderGraph, HDCamera hdCamera, Textu
                         passData.pingFarRGB = builder.CreateTransientTexture(GetPostprocessOutputHandle(renderGraph, "DoF Source Pyramid", GetPostprocessTextureFormat(), true));
 
                         float scaleFactor = 1.0f / passData.parameters.minMaxCoCTileSize;
-                        passData.pingNearRGB = builder.CreateTransientTexture(new TextureDesc(Vector2.one * scaleFactor, true, true)
+                        passData.pingNearRGB = builder.CreateTransientTexture(new TextureDesc(Vector2.one * scaleFactor, IsDynamicResUpscaleTargetEnabled(), true)
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.PostProcess.cs:2882]: This as well was for DLSS too I think, but only in HW mode (cc @kecho)

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs [hunk 13]
```diff
@@ -145,9 +179,17 @@ static void Drawer_Rendering_Antialiasing_SMAA(SerializedHDCamera p, Editor owne
...
 
+            static void Draw_Rendering_Antialiasing_TAA_Advanced(SerializedHDCamera p, Editor owner)
+            {
+                EditorGUILayout.PropertyField(p.taaBaseBlendFactor, Styles.TAABaseBlendFactor);
+            }
+
             static void Drawer_Rendering_Antialiasing_TAA(SerializedHDCamera p, Editor owner)
             {
                 EditorGUILayout.PropertyField(p.taaQualityLevel, Styles.TAAQualityLevel);
+                if (s_IsRunningTAAU)
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Rendering.Drawers.cs:199]: (See previous comment image for how it looks now)

**Comment by FrancescoC-unity** [comment]
Fixed few issues and addressed review comments

**Review by remi-chapelain — CHANGES_REQUESTED** [review]
In addition to the **performance tests** I already did in the PR description and the general visual pass on the quality of the upsample pair-testing with @FrancescoC-unity,
 
### Here's what I tested:
- Changing resolution when using the filter ❌ 
- Tested with DLSS that it disabled correctly the Fallback post AA ✔️ 
- Tampering with HDRP Asset when using the filter ✔️ 
- Tested behavior with VFX Graph and moving objects in general ✔️  
- Disabling custom frame settings (especially post process and AA) ✔️ 
- Made a small script to modulate screen percentage to stay at a constant FPS at 4k ✔️ 
- Various post process effects like DoF, Motion blur, Fog ⚠️  
- MSAA at various level ✔️ 
- Forward / Deferred ✔️ 
- Color buffer and PP buffer format R11G11B10 and R16G16B16A16 ✔️ 

Also, couldn't test building and changing resolution there because of [this recent issue](https://fogbugz.unity3d.com/f/cases/1349162/) in trunk

### Issues founds:
- When you have fog in the scene, enabling DRS with TAAU filter gives you a flash.  **[FIXED]**
https://user-images.githubusercontent.com/57442369/125479978-56fb83c8-212f-47c2-aba7-f63ee9613a97.mp4

- When you disable post process (specifically AA), the frame is reduced to whatever force percentage is set in your HDRP asset and the rest is black. I think @kecho mentioned
 it also a bit above.  **[FIXED]**
https://user-images.githubusercontent.com/57442369/125480108-bba090f6-a05c-4984-b26c-d7ce54242762.mp4

- Going to a higher resolution while the screen percentage is low results in some artefacts outside the low buffer resolution.
https://user-images.githubusercontent.com/57442369/125480243-95981f04-7fbd-41a3-8d53-a0dad08795c3.mp4


### Few concerns :
- **Documentation**, I won't block this PR for now, but now that we have a bunch of these filters, it can be useful for users to have some sort of pros and cons of each in the documentation to help them pick a filter depending on the context. For now, in the [DRS HDRP page](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@12.0/manual/Dynamic-Resolution.html), we don't talk about any of them. 
- **Base Blend factor**, If I understood it correctly, this is the same type of parameter as we have in Volumetric Clouds and SSR with PBR Algorithm. 
In those effects, they are called "Temporal Accumulation" and "Accumulation Factor" respectively. If they do basically the same thing, could we unify this term to make it clear to users that it's the same principle behind ?

**Comment by FrancescoC-unity** [comment]
> * When you disable post process (specifically AA), the frame is reduced to whatever force percentage is set in your HDRP asset and the rest is black. I think @kecho mentioned
>   it also a bit above.
>   https://user-images.githubusercontent.com/57442369/125480108-bba090f6-a05c-4984-b26c-d7ce54242762.mp4
> *

Nope different issue, the one mentioned by Kleber got since fixed. 
We discussed this, can be easily fixed will do when I get the chance. 

> * When you have fog in the scene, enabling DRS with TAAU filter gives you a flash.
>   https://user-images.githubusercontent.com/57442369/125479978-56fb83c8-212f-47c2-aba7-f63ee9613a97.mp4
> *

Probably due to history of exposure being reset as well, I vaguely remember this getting fixed by another PR. It is something I think we should see if still there after I merge master and if it is file a separate bug related to it. 

> * Going to a higher resolution while the screen percentage is low results in some artefacts outside the low buffer resolution.
>   https://user-images.githubusercontent.com/57442369/125480243-95981f04-7fbd-41a3-8d53-a0dad08795c3.mp4

This is instead just very bad :D Will fix. 

> * **Documentation**, I won't block this PR for now, but now that we have a bunch of these filters, it can be useful for users to have some sort of pros and cons of each in the documentation to help them pick a filter depending on the context. For now, in the [DRS HDRP page](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@12.0/manual/Dynamic-Resolution.html), we don't talk about any of them.

As I mentioned in the PR, we do definitively need Docs, this PR however is likely to land with the option hidden as I need to sort the Exclude from TAA issue separately. Will likely write docs when I have the chance to come back to this PR in full force. 


> * **Base Blend factor**, If I understood it correctly, this is the same type of parameter as we have in Volumetric Clouds and SSR with PBR Algorithm.
>   In those effects, they are called "Temporal Accumulation" and "Accumulation Factor" respectively. If they do basically the same thing, could we unify this term to make it clear to users that it's the same principle behind ?

Nope, fairly different. TAA blend factor is more complex than those two in that it is modified by external parameters too instead of been used as is. I think they are still fairly different and I think similarly that the word _base_ is really important here given that the value will likely be changed significantly per pixel.

**Comment by FrancescoC-unity** [comment]
@remi-chapelain  I fixed the white flash and the black screen, can't repro the other issue, can you take another look when you get the chance?

**Review by remi-chapelain — APPROVED** [review]
Made another quick pass and updated my previous comment, one more issue was found and fixed and the rest seems good for shipping to me ! 

Only issue that isn't fixed is [the artefact when the resolution is changed](https://unity.gyazo.com/2c7d080831e24b9445f63251998dbbb7) (from low to high) when DRS is already very low not in forced percentage mode. 
I reported [an linked issue](https://fogbugz.unity3d.com/f/cases/1352848/) with DLSS that will hopefully fix it later. 

I'd like documentation to be updated with pros and cons about other filter to help user picking but this can happen in a second pass on DRS as a whole.

**Comment by FrancescoC-unity** [comment]
Yup, documentation will happen as soon as this land, there is also a bit of guideline needed for TAAU specifically so will write it later.

**Review by sebastienlagarde — APPROVED** [review]
